compiler/sem/semtypes.nim

```diff
@@ -494,7 +504,13 @@ proc semAnonTuple(c: PContext, n: PNode, prev: PType): PType =
     localReport(c.config, n, reportSem rsemTypeExpected)
   result = newOrPrevType(tyTuple, prev, c)
   for it in n:
-    addSonSkipIntLit(result, semTypeNode(c, it, nil), c.idgen)
+    let t = semTypeNode(c, it, nil).skipIntLit(c.idgen)
+    var cycleDetector = initIntSet()
+    if isRecursiveType(t, cycleDetector):
+      result.add t
+      c.config.localReport(n.info, reportTyp(rsemIllegalRecursion, t))
+    else:
+      rawAddSon(result, t)
```

Comment:
The general direction of the suggestion above also applies here.

Reply:
Oops, sorry. The comments got reordered when submitting the review. [This](https://github.com/nim-works/nimskull/pull/716#discussion_r1207140928) is the suggestion I was referring to.

Reply:
Updated